ls1intum/Artemis
# [Enhancement] Validation for Unreferenced Feedback of Programming Exercises

### Checklist


### Motivation and Context


We want to visualise that the unreferenced feedback should have a score and a detailed text. This is needed specially for programming exercises.

### Description

- Added `required` to `<input>` and `<textarea>` DOM elements
- Changed the check when it is allowed to save/submit a assessment for programming exercises
### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Assess a programming exercise
3.1 Add a referenced (inline) feedback -> You should be able to submit/save (see if buttons are disabled). Delete this feedback afterwards
3.2 Add unreferenced feedback -> You should not be able to submit/save unless you entered a detail text and score. Delete this feedback afterwards
3.3 Add general feedback -> You should be able to submit/save.
3.4 Add a referenced (inline) feedback -> You should be able to submit/save.
3.5 Add unreferenced feedback -> You should not be able to submit/save unless you entered a detail text and score. 
3.6 Add more unreferenced feedback -> You should not be able to submit/save unless you entered a detail text and score. 

### Test Coverage

Nothing changed.

### Screenshots


Unreferenced feedback 
![image](https://user-images.githubusercontent.com/41108487/98467242-88fd0a00-21d4-11eb-8ebc-c598cc960a68.png)
